### ISSUE TITLE
RNNs papers to read

### ISSUE THREAD

**Issue by ih4cku** [ih4cku/blog#97]
# ablation studies of gates
- Chung, Junyoung, et al. "Empirical evaluation of gated recurrent neural networks on sequence modeling." arXiv preprint arXiv:1412.3555 (2014).
- Greff, Klaus, et al. "LSTM: A search space odyssey." arXiv preprint arXiv:1503.04069 (2015).

**Comment by ih4cku** [comment]
# Make implementation fast
- Baidu Silicon Valley AI Lab http://svail.github.io/

**Comment by ih4cku** [comment]
# vanishing and exploding gradient / sensitivity
- (**must see**) Pascanu, Razvan, Tomas Mikolov, and Yoshua Bengio. "On the difficulty of training recurrent neural networks." ICML (3) 28 (2013): 1310-1318.
- [Written Memories: Understanding, Deriving and Extending the LSTM](http://r2rt.com/written-memories-understanding-deriving-and-extending-the-lstm.html)
- [Why are deep neural networks hard to train?](http://neuralnetworksanddeeplearning.com/chap5.html#the_vanishing_gradient_problem)
- [Why can Constant Error Carousels (CECs) prevent LSTM from the problems of vanishing/exploding gradients?](https://www.reddit.com/r/MachineLearning/comments/34piyi/why_can_constant_error_carousels_cecs_prevent/)

**Comment by ih4cku** [comment]
# BPTT
- Section 2.8.6 of Ilya Sutskever’s [Ph.D. thesis](http://www.cs.utoronto.ca/~ilya/pubs/ilya_sutskever_phd_thesis.pdf)
- [Styles of Truncated Backpropagation](http://r2rt.com/styles-of-truncated-backpropagation.html)
